kayenta-datadog/src/main/java/com/netflix/kayenta/datadog/metrics/DatadogMetricsService.java

```diff
@@ -79,8 +80,27 @@ public String buildQuery(
       CanaryScope canaryScope) {
     DatadogCanaryMetricSetQueryConfig queryConfig =
         (DatadogCanaryMetricSetQueryConfig) canaryMetricConfig.getQuery();
+    String metricName = queryConfig.getMetricName();
+    String scope = canaryScope.getScope();
+    List<String> tags = queryConfig.getTags();
+    if (queryConfig.getMetricType() == MetricType.COUNT) {
+      return composeCountQuery(metricName, scope, tags);
+    }
+    // Currently, treat Rate and Gauge the same.
+    return composeGaugeQuery(metricName, scope, tags);
+  }
 
-    return queryConfig.getMetricName() + "{" + canaryScope.getScope() + "}";
+  private String composeCountQuery(String metricName, String scope, List<String> tags) {
+    String gaugeQuery = composeGaugeQuery(metricName, scope, tags);
+    return "sum:" + gaugeQuery + ".as_count()";
```

Comment:
@dotdotdotpaul Unfortunately the `count` prefix seems to still apply smoothing:

![image](https://user-images.githubusercontent.com/4308672/63984331-9de1f780-ca7f-11e9-9c1b-b7c35588fea0.png)

I do agree we have some deviation between the frontend prefix and the backend string formatting. To be honest, I wasn't aware you were working on this at the same time. Given that we're both invested in getting this right, perhaps we can sync up and align on a long term vision for these datadog metric aggregations? :)

Reply:
The only thing I worked on recently was fixing the UI so you can put the aggregator prefix in (avg, min, max, sum) because _originally_, that field was just a text box so you would type it in yourself, directly, ala "avg:system.cpu.used" or whatever.  Someone made a nice feature to do the metric lookup -- but then you couldn't select/enter the aggregator.  People had to edit the JSON directly to make it work.  So I fixed _that_.  Nothing else.  😄

Reply:
If you remove the extra "sum:" from the front of that string (or remove whatever aggregator was selected in favor of "sum:", when doing the .as_count() thing), I'll sign off on this.  I don't know the details of Datadog's metrics reporting well enough to say "no" -- But I do think that prepending of "sum:" is a bug.  And then, on the UI side, there's the question of how this "as_count" option will be selectable/recorded.  But backend first, so long as it's compatible with the existing JSON configs, is where to start.

Reply:
@dotdotdotpaul 

I agree that in its current state we can't merge this backend change in - it will lead to an incompatibility with the frontend.

How would you feel about the following:

- I update this backend change to accept "space-aggregation" and "time-aggregation" components, along with the tag strings. Together, these component strings will cover everything needed for a full-fledged DD query.
- We couple the backend change with a frontend change allowing users to specify tags, and both aggregations. We store the aggregations as fields in the Canary Metric Config rather than attaching them to the metric name.

I feel like this would build on the spirit of your frontend change while providing users with a fuller range of datadog queries. 

If you like this idea, I can consider building it internally at Airbnb and later upstream it. We could close this existing PR for now.

Reply:
I'm all for extending it to be cleaner.  Backwards compatibility is a concern/question, but that's solvable.  I don't think we're too far off here, I really think your change here is fine so long as the extra prepended "sum:" is removed.  😄 Without a UI to indicate the "as_count" field, nobody will see anything different, unless they edit the JSON directly.  We can then iterate on the UI after that, maybe refactor a little of this then.